foreman/data_refinery_foreman/surveyor/utils.py

```diff
@@ -43,6 +44,29 @@ def flatten(d, parent_key="", sep="_"):
     return dict(items)
 
 
+# TODO: Merge this with above method after updating array express
+#       to support recursive flattening.
+def flatten_dict(data: Union[Dict, List], prefix: str = "", sep="_") -> dict:
+    """Flattens a nested dictionary or list"""
+    flattened = {}
+    iterator = enumerate(data) if isinstance(data, list) else data.items()
+
+    for k, v in iterator:
+        key = (k if not prefix else f"{prefix}{sep}{k}").replace(" ", sep)
```

Comment:
Yeah thats fair, I think this is safe because of how we are using it. I don't think we actually need the separator argument but I kept it because I wanted more overlap of functionality when we go to merge it with the other non-recursive flatting utility method. The collision here wouldn't really matter with how the harmonizer works as it checks spaces, dashes, and underscores regardless. It is harder to work out which spaces were nested keys, but in all honesty we don't really care we just need keys to be unique. When I implement array express and then merge these functions I will probably remove it in favor of a more opinionated approach.